### DISCUSSION TITLE
Wrong ed25519 keys?

### DISCUSSION THREAD

**Discussion by master255** [arvidn/libtorrent#7527] [Q&A]
@arvidn The ed25519 keys in library are not the same as the ed25519 keys used around the world. Is this some kind of its own version of the keys?
Throughout the world, a private key is 32 bytes of the seed + 32 bytes of the public key. And [here the](https://github.com/arvidn/libtorrent/blob/5ce4c516479085eda9327d35891b7732d6d5037b/src/ed25519/keypair.cpp#L14) private key is a 64 byte hash of the seed. Probably you need to rename ed25519 to ed25519t at least? To avoid confusion?

And of course, worldwide ed25519 keys do not pass verification in the ed25519 torrent keys and vice versa. These are different keys. Although they have the same name and the same size.

How keys are generated worldwide: https://github.com/golang/go/blob/1e690409206ff97330b5a91517d453fc5129bab2/src/crypto/ed25519/ed25519.go#L172C43-L172C43
This matches Bouncycastle keys.

**Comment by arvidn** [comment]
this is the document your comment seem to apply to: http://bittorrent.org/beps/bep_0044.html
you can bring up your concern about naming on the bittorrent.org github repository
